test/rulesets/Base/arraymath.jl

```diff
@@ -15,12 +15,27 @@
         ⋆₂(a, b) = ⋆₂((a, b))  #matrix
         ⋆₂() = ⋆₂(())  # scalar
 
+        ⋆₃(a) = (⋆(n), ⋆(n, 1))
+
         @testset "Scalar-Array $dims" for dims in ((3,), (5,4), (2, 3, 4, 5))
             rrule_test(*, ⋆(dims), ⋆₂(), ⋆₂(dims))
             rrule_test(*, ⋆(dims), ⋆₂(dims), ⋆₂())
         end
 
         @testset "AbstractMatrix-AbstractMatrix" begin
+            # Matrix-Vector product
+            @testset "n=$n, m=$m" for n in (2, 3), m in (4, 5)
+                @testset "Array" begin
+                    rrule_test(*, ⋆(n), ⋆₂(n, m), ⋆₂(m))
+                end
+            end
+
+            @testset "n=$n, m=$m" for n in (2, 3), m in (4, 5)
+                @testset "Array" begin
+                    rrule_test(*, ⋆(n, m), ⋆₃(n), ⋆₂(1, m))
+                end
+            end
```

Comment:
We should make sure we haven't broken Covector-vector products
```suggestion
            end
            
            @testset "Covector * Vector n=$n" for in in (3, 5)
                @testset "$f" for f in (adjoint, transpose)
                    # This should be same as dot product and give a scalar
                    rrule_test(*, ⋆(), f.(⋆₂(n)), ⋆₂(n))
                end
            end
```

Reply:
These tests will probably pass.

Though, I'm not really sure of the meaning of Co-vector, can you please point me to the correct wiki article? Is below the correct definition (taken from http://www.physicsinsights.org/pbp_one_forms.html)
```
 A "covector" is a rank (0,1) tensor, also called a "dual vector". 
It is a linear function which maps vectors into real numbers. 
The term "dual vector" is actually more popular than "covector" but seems cumbersome to me. 
"Covector" also suggests "covariant vector", in contrast to ordinary
vectors, which are "contravariant".
```

Reply:
Covector is `Adjoint{<:Vector}` and `Transpose{<: Vector}`

Reply:
we do need these tests